addons/uom/static/src/components/many2x_uom_tags/many2x_uom_tags.js

```diff
@@ -0,0 +1,81 @@
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { Many2XAutocomplete } from "@web/views/fields/relational_utils";
+import {
+    Many2ManyTagsFieldColorEditable,
+    many2ManyTagsFieldColorEditable,
+} from "@web/views/fields/many2many_tags/many2many_tags_field";
+import {
+    Many2OneField,
+    many2OneField,
+} from "@web/views/fields/many2one/many2one_field";
+import { UomAutoComplete } from "@uom/components/uom_autocomplete/uom_autocomplete";
+
+export class Many2XUomTagsAutocomplete extends Many2XAutocomplete {
+    static components = {
+        ...Many2XAutocomplete.components,
+        AutoComplete: UomAutoComplete,
+    };
+
+    async search(name) {
+        const records = await this.orm.call(
+            this.props.resModel, "search_read", [], {
+            domain: [...this.props.getDomain(), ["name", "ilike", name]],
+            fields: ["id", "name", "relative_factor", "relative_uom_id"],
+        });
+        const relativeUomIds = records.filter((record) => record.relative_uom_id).map((record) => record.relative_uom_id[0]);
```

Comment:
Since you now have a `parent_path` on the uoms, can't you use that to fetch all relevant uoms in a single call?